eng/common/pipelines/templates/steps/save-package-properties.yml

```diff
@@ -32,6 +32,19 @@ steps:
             -ArtifactPath '${{ parameters.DiffDirectory }}'
           pwsh: true
 
+      # When running in PR node, we want the detected changed services to be attached to the build as tags.
+      # However, the public identity does not have the permissions to attach tags to the build.
+      # Instead, we will save the changed services to a file, attach it as an attachment for PiplineWitness to pick up and utilize.
+      - pwsh: |
+          $changedServices = (Get-Content -Path '${{ parameters.DiffDirectory }}/diff.json' -Raw | ConvertFrom-Json).ChangedServices
+
+          if ($changedServices) {
+            Write-Host "Attaching changed service names to the build for additional tag generation."
```

Comment:
It might also be worth logging the services in the logs as well.

Reply:
We dump the PR diff value in `Generate-PR-Diff.ps1`. I think I'm going to leave it alone here if that's ok with you.

Reply:
That is fine. Is there a reason you don't just add the tags attachment in that step as well?

Reply:
I did absolutely think about putting it there. I chose not to because

A) We normally want to keep our scripts to single purpose. Parsing the package info, generating the PR diff, and writing that json out is a single complete task. This attachment is secondary.
B) The template offers surety that it will run if applicable, and if anything goes wrong during upload it'll be a separate step to easily highlight.

To be entirely honest, I don't have a preference other than what reasoning I gave above.

Reply:
I also don't have a strong preference but wanted to see if you considered it. I'm good with this being an independent step.